csharp/IceDiscovery/Replication/Client/Program.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) ZeroC, Inc.
+
+// Slice module VisitorCenter in Greeter.ice maps to C# namespace VisitorCenter.
+using VisitorCenter;
+
+// Configure the communicator to load the IceDiscovery plugin during initialization. This plugin installs a default
+// locator on the communicator.
+var initData = new Ice.InitializationData
+{
+    properties = new Ice.Properties(ref args)
+};
+initData.properties.setProperty("Ice.Plugin.Discovery", "IceDiscovery:IceDiscovery.PluginFactory");
+
+// Create an Ice communicator to initialize the Ice runtime. The communicator is disposed before the program exits.
+using Ice.Communicator communicator = Ice.Util.initialize(initData);
+
+// Create proxy to the Greeter object hosted by the server. "greeter" is a stringified proxy with no addressing
+// information, also known as a well-known proxy. It's resolved by the default locator installed by the IceDiscovery
+// plugin.
+GreeterPrx greeter = GreeterPrxHelper.createProxy(communicator, "greeter");
+
```

Comment:
One issue with these IceDiscovery demos is that we assume the locator was set at this point. If I recall correctly the plug-in does this in the background, and there is no API to wait for it.

Reply:
Doesn't it do in during plugin init?

Reply:
I am not aware of any issue or bug with the IceDiscovery plugin initialization.